src/test/java/org/prebid/server/bidder/applogy/ApplogyBidderTest.java

```diff
@@ -0,0 +1,293 @@
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.banner;
+import static org.prebid.server.proto.openrtb.ext.response.BidType.video;
+
+public class ApplogyBidderTest extends VertxTest {
+
+
+    private static final String ENDPOINT_URL = "https://test.endpoint.com";
+
+    private ApplogyBidder applogyBidder;
+
+    @Before
+    public void setUp() {
+        applogyBidder = new ApplogyBidder(ENDPOINT_URL);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new ApplogyBidder("invalid_url"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldSkipEmptyTokens() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpApplogy.of(null)))));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = applogyBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput(
+                        "token is empty"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldSkipInvalidImpressionAndAddError() {
+        // given
+        ExtPrebid<?, ExtImpApplogy> ext = ExtPrebid.of(null, ExtImpApplogy.of("token1"));
+        Imp imp = givenImp(
+                impBuilder -> impBuilder
+                        .banner(null)
+                        .id("2")
+                        .ext(mapper.valueToTree(ext))
+                        .banner(Banner.builder().w(300).h(400).build())
+        );
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(
+                        imp,
+                        givenImp(impBuilder -> impBuilder
+                                .banner(null)
+                                .id("2")
+                                .ext(mapper.valueToTree(ext))
+                                .audio(Audio.builder().build()))))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = applogyBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput(
+                        "Applogy only supports banner, video or native ads"));
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .flatExtracting(BidRequest::getImp)
+                .containsOnly(imp);
+    }
+
+
+    @Test
+    public void makeBidsShouldReturnErrorWhenResponseBodyCouldNotBeParsed() {
```

Comment:
This should be placed after all `makeHttpRequests` method cases. Please ensure the correct tests order.